src/Services/Products/Distribt.Services.Products.Api.Read/Program.cs

```diff
@@ -1,9 +1,17 @@
 using Distribt.Services.Products.BusinessLogic.DataAccess;
+using Distribt.Shared.Databases.MongoDb;
+using Microsoft.Extensions.Options;
 
-WebApplication app = DefaultDistribtWebApplication.Create(args, builder =>
+WebApplication app = await DefaultDistribtWebApplication.Create(args, async builder =>
 {
-    builder.Services.AddDistribtMongoDbConnectionProvider(builder.Configuration)
-        .AddScoped<IProductsReadStore, ProductsReadStore>();
+    (await builder.Services.AddDistribtMongoDbConnectionProvider(builder.Configuration))
+        .AddScoped<IProductsReadStore>((IServiceProvider serviceProvider) =>
+        {
+            var mongoDbConnectionProvider = serviceProvider.GetService<IMongoDbConnectionProvider>() ?? throw new ArgumentNullException();
```

Comment:
yep here it is the other comment, as this is not abstracted away this is being done every time project you need mongo.